File: app/src/main/java/com/example/app_tema1_fredpor/inicio.java
package com.example.app_tema1_fredpor;


import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.RadioGroup;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class inicio extends AppCompatActivity {
    private boolean checkRadio = false;
    private boolean checkCheckbox = false;
    private static Usuario usuario;
    private static String userName;
    private static String userPass;
    private static String userSex = "";

    @Override
    protected void  onCreate (Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.inicio);


        Button botonStart = findViewById(R.id.botonEntrar);
        RadioGroup radioGroup = findViewById(R.id.radioGroup);
        CheckBox checkBox = findViewById(R.id.check_condiciones);

        EditText textUser = findViewById(R.id.textUser);
        EditText textPass = findViewById(R.id.textPassword);





        final boolean[] checkCheckBox = {false};

        radioGroup.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup radioGroup, int i) {
                checkRadio = i != -1;
            }
        });

        checkBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton compoundButton, boolean b) {
                checkCheckbox = b;
            }
        });

        botonStart.setOnClickListener(new View.OnClickListener() {
            @SuppressLint("ResourceType")
            @Override
            public void onClick(View v) {
                userName = textUser.getText().toString();
                userPass = textPass.getText().toString();
                if (checkCheckbox && checkRadio){
                    if(!TextUtils.isEmpty(userName) && !TextUtils.isEmpty(userPass)){
                        int radioCheckedId = radioGroup.getCheckedRadioButtonId();
                        switch (radioCheckedId){
                            case 2131230828: //id del boton1
                                userSex = "H";
                                break;
                            case 2131230829://id del boton2
                                userSex = "M";

                                break;
                            case 2131230830://id del boton3
                                userSex = "O";
                                break;
                            default:
                                Toast.makeText(getApplicationContext(), radioGroup.getCheckedRadioButtonId(), Toast.LENGTH_SHORT).show();
                        }
                        Intent intent = new Intent(inicio.this, menu.class);
                        startActivity(intent);
                    }else{
                        Toast.makeText(getApplicationContext(), "DEBES INTRODUCIR USUARIO Y CONTRASEÑA", Toast.LENGTH_SHORT).show();
                    }

                }else{
                    Toast.makeText(getApplicationContext(), "DEBES SELECCIONAR EL SEXO Y ACEPTAR LAS CONDICICONES", Toast.LENGTH_SHORT).show();
                }
            }
        });

    }
    public static Usuario getUsuario(){
        return usuario = new Usuario(userName, userPass, userSex);
    }


}

File: app/src/main/java/com/example/app_tema1_fredpor/Inicio.kt
package com.example.app_tema1_fredpor

import android.content.Intent
import android.os.Bundle
import android.text.TextUtils
import android.util.Log
import android.widget.Button
import android.widget.CheckBox
import android.widget.EditText
import android.widget.RadioGroup
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class Inicio : AppCompatActivity() {
    private var checkRadio = false
    private var checkCheckbox = false
    private lateinit var textUser: EditText
    private lateinit var textPass: EditText
    private var userSex = ""

    override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)
        setContentView(R.layout.inicio)

        val botonStart = findViewById<Button>(R.id.botonEntrar)
        val radioGroup = findViewById<RadioGroup>(R.id.radioGroup)
        val checkBox = findViewById<CheckBox>(R.id.check_condiciones)

        textUser = findViewById(R.id.textUser)
        textPass = findViewById(R.id.textPassword)

        radioGroup.setOnCheckedChangeListener { _, i ->
            checkRadio = i != -1
        }

        checkBox.setOnCheckedChangeListener { _, b ->
            checkCheckbox = b
        }

        botonStart.setOnClickListener {

            var userName = textUser.text.toString()
            var userPass = textPass.text.toString()
            if (checkCheckbox && checkRadio) {
                if (!TextUtils.isEmpty(userName) && !TextUtils.isEmpty(userPass)) {
                    val radioCheckedId = radioGroup.checkedRadioButtonId
                    when (radioCheckedId) {
                        R.id.btr_hombre -> {
                            userSex = "H"
                            Log.e("TAG", "Este es un mensaje de error")
                        }
                        R.id.btr_mujer -> {
                            userSex = "M"
                            Log.e("TAG", "Este es un mensaje de error")
                        }
                        R.id.btr_otro -> {
                            userSex = "O"
                            Log.e("TAG", "Este es un mensaje de error")
                        }
                        else -> {
                            Toast.makeText(applicationContext, "Error, los id han cambiado", Toast.LENGTH_SHORT).show()
                        }
                    }
                    ListarUser.annadirUsuario(Usuario(userName,userPass,userSex))
                    val intent = Intent(this@Inicio, Menu::class.java)
                    startActivity(intent)
                } else {
                    Toast.makeText(applicationContext, "DEBES INTRODUCIR USUARIO Y CONTRASEÑA", Toast.LENGTH_SHORT).show()
                }
            } else {
                Toast.makeText(applicationContext, "DEBES SELECCIONAR EL SEXO Y ACEPTAR LAS CONDICICONES", Toast.LENGTH_SHORT).show()
            }
        }
    }

}